src/lib/drivers/driver.js

```diff
@@ -250,7 +256,7 @@ class DriverBase {
 
         if (response.eof) {
           isEOF = true;
-          resolve(JSON.parse(result));
+          return resolve(JSON.parse(result));
```

Comment:
Saw some occasions where we continue to send IO.read even after EOF.  Feel much better about quitting in this case.